ci-operator/templates/openshift/qe/cluster-launch-installer-qe.yaml

```diff
@@ -0,0 +1,360 @@
+            elif oc get deploy/router -n openshift-ingress 2>/dev/null; then
+              ROUTER_NAMESPACE=openshift-ingress
+              ROUTER_DEPLOYMENT="deploy/router"
+            elif oc get deploy/router -n default 2>/dev/null; then
+              ROUTER_NAMESPACE=default
+              ROUTER_DEPLOYMENT="deploy/router"
+            else
+              echo "Waiting for router to be created ..."
+              sleep 15 & wait
+              continue
+            fi
+            echo "Found router in ${ROUTER_NAMESPACE}"
+          fi
+          break
+        done
+
+        i=0
+        MAX_RETRIES=10
+        wait_expiry_ts="$(($(date +%s) + 90))"
+        until oc --request-timeout=80s rollout status "${ROUTER_DEPLOYMENT}" -n "${ROUTER_NAMESPACE}" -w; do
+            i=$((i+1))
+            [ $i -eq $MAX_RETRIES ] && echo "timeout waiting for ${ROUTER_NAMESPACE}/${ROUTER_DEPLOYMENT} to be available" && exit 1
+            echo "error ${ROUTER_NAMESPACE}/${ROUTER_DEPLOYMENT} did not come up"
+            sleep "$((wait_expiry_ts - $(date +%s)))"
+            wait_expiry_ts="$(($(date +%s) + 90))"
+        done
+
+        # set up cloud-provider-specific env vars
+        if [[ "${CLUSTER_TYPE}" == "gcp" ]]; then
+          export GOOGLE_APPLICATION_CREDENTIALS="/tmp/cluster/gce.json"
+          export KUBE_SSH_USER=cloud-user
+          mkdir -p ~/.ssh
+          cp /tmp/cluster/ssh-privatekey ~/.ssh/google_compute_engine || true
+          export PROVIDER_ARGS='-provider=gce -gce-zone=us-east1-c -gce-project=openshift-gce-devel-ci'
+        elif [[ "${CLUSTER_TYPE}" == "aws" ]]; then
+          export PROVIDER_ARGS="-provider=aws -gce-zone=us-east-1"
+          mkdir -p ~/.ssh
+          cp /tmp/cluster/ssh-privatekey ~/.ssh/kube_aws_rsa || true
+        elif [[ "${CLUSTER_TYPE}" == "openstack" ]]; then
+          mkdir -p ~/.ssh
+          cp /tmp/cluster/ssh-privatekey ~/.ssh/kube_openstack_rsa || true
+        fi
+
+        set -x
+        # Configure allowAll authentication
+        oc patch clusterversions.config.openshift.io/version -p '{"spec":{"overrides": [{"kind": "Deployment","name": "openshift-cluster-kube-apiserver-operator","unmanaged": true}]}}' --type merge
+        oc patch kubeapiserveroperatorconfig instance -p '{"spec":{"unsupportedConfigOverrides": {"oauthConfig": {"identityProviders": [{"challenge": true,"login": true,"name": "anypassword","provider": {"apiVersion": "v1","kind": "AllowAllPasswordIdentityProvider"}}],"masterCA": "/etc/kubernetes/static-pod-resources/configmaps/kubelet-serving-ca/ca-bundle.crt","masterPublicURL": "https://${OPENSHIFT_INSTALL_CLUSTER_NAME}.${OPENSHIFT_INSTALL_BASE_DOMAIN}:6443","masterURL": "https://${OPENSHIFT_INSTALL_CLUSTER_NAME}.${OPENSHIFT_INSTALL_BASE_DOMAIN}:6443"}}}}' --type merge
+        set +x
+
+        # wait for changes to be applied.
+        sleep 300
+
+        /usr/bin/scl enable rh-git29 rh-ror50 -- ${TEST_COMMAND}
```

Comment:
What's the diff here between this template and the `cluster-launch-src`?

Reply:
I configure the cluster with allowall authentication above. They will need more configuration in the future.

Reply:
Can we move that to their scripts in their repo? Why do we need a separate template? They have `oc` and the admin kubeconfig, they can run the  commands in their test script setup?

Reply:
There are also some env vars that they need to be configured. For [example](https://github.com/openshift/release/blob/3759437db1e654c238d9607ea03b542853f994ae/ci-operator/templates/openshift/qe/cluster-launch-installer-qe.yaml#L97). This varable depends on the `${NAMESPACE}` and `${JOB_NAME_HASH}` variables. Also there are more that they need to be define.  I will test if we can define them in the ${TEST_COMMAND} as usual.

Reply:
And If we can find a way to pass the env vars correct using only the `${TEST_COMMAND}`, their test command will be look like:
```yaml
      - name: TEST_COMMAND
        value: |
          export OPENSHIFT_ENV_OSE_HOSTS=${NAMESPACE}-${JOB_NAME_HASH}-api.origin-ci-int-aws.dev.rhcloud.com:master:etcd:node; \
          export BUSHSLICER_DEFAULT_ENVIRONMENT=ose; \
          export OPENSHIFT_ENV_OSE_USER_MANAGER_USERS=test:test; \
          export OPENSHIFT_ENV_OSE_API_PORT="6443"; \
          oc patch clusterversions.config.openshift.io/version -p '{"spec":{"overrides": [{"kind": "Deployment","name": "openshift-cluster-kube-apiserver-operator","unmanaged": true}]}}' --type merge; \
          oc patch kubeapiserveroperatorconfig instance -p '{"spec":{"unsupportedConfigOverrides": {"oauthConfig": {"identityProviders": [{"challenge": true,"login": true,"name": "anypassword","provider": {"apiVersion": "v1","kind": "AllowAllPasswordIdentityProvider"}}],"masterCA": "/etc/kubernetes/static-pod-resources/configmaps/kubelet-serving-ca/ca-bundle.crt","masterPublicURL": "https://${OPENSHIFT_INSTALL_CLUSTER_NAME}.${OPENSHIFT_INSTALL_BASE_DOMAIN}:6443","masterURL": "https://${OPENSHIFT_INSTALL_CLUSTER_NAME}.${OPENSHIFT_INSTALL_BASE_DOMAIN}:6443"}}}}' --type merge; \
          sleep 300; \
          /usr/bin/scl enable rh-git29 rh-ror50 -- cucumber --name "Could not create any context in non-existent project"
```
and of course `export OPENSHIFT_ENV_OSE_HOSTS=${NAMESPACE}-${JOB_NAME_HASH}-api.origin-ci-int-aws.dev.rhcloud.com:master:etcd:node` cannot be initialized. Also, the command `oc patch kubeapiserveroperatorconfig` will fail for the same reason.

Reply:
Why can't we add that as a script to their repo and have the command run it?

Reply:
@stevekuznetsov working on this now for a related PR, thanks: https://github.com/openshift/release/pull/2236

Reply:
PR for script: [#37](https://github.com/openshift/verification-tests/pull/37)